report/report.py

```diff
@@ -0,0 +1,98 @@
+from sisyphus import *
+
+import getpass
+import gzip
+from typing import Dict, Union, Callable, Optional
+import pprint
+
+_Report_Type = Dict[str, Union[tk.AbstractPath, str]]
+
+
+class GenerateReportStringJob(Job):
+    """
+    Job to generate and output a report string
+    """
+
+    def __init__(
+        self,
+        report_values: Union[_Report_Type, Callable],
+        report_template: Optional[Callable[[_Report_Type], str]] = None,
+        compress: bool = True,
+    ):
+        """
+
+        :param report_values:
+        :param report_template:
+        :param compress:
+        """
+        self.report_values = report_values
+        self.report_template = report_template
+        self.compress = compress
+
+        self.out_report = self.output_path(
+            "report.txt.gz" if self.compress else "report.txt"
+        )
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+
+        if self.report_template:
+            report = self.report_template(self.report_values)
```

Comment:
I think it would be nice if `report_values` and `report_template` follows the same interface as `register_report` in sisyphus. This allows to reuse existing "sisyphus reports" with these jobs to send mails.
In sisyphus report_template is not a callable but a format string (see https://github.com/rwth-i6/sisyphus/blob/032239c4f473742b96178f9614521324ac0472f7/sisyphus/graph.py#L187).

Reply:
I would like to support both, but added the option if the type of the `report_template` is str to be used as format string.